node/src/blockchain/blockchain_interface.rs

```diff
@@ -718,14 +801,18 @@ mod tests {
             .into_iter()
             .map(|request| serde_json::from_slice(&request.body()).unwrap())
             .collect();
+        let bodies: Vec<String> = bodies
```

Comment:
Why bother to `.collect()` the first iterator? Why not just have two adjacent `.map()` calls and one `.collect()`? That way you wouldn't have two differently-typed `bodies` variables and it wouldn't be as confusing to read.

Reply:
ok I can combine them.